synapse/lib/httpapi.py

```diff
@@ -597,18 +691,20 @@ async def post(self):
         name = body.get('user')
         passwd = body.get('passwd')
 
-        user = await self.cell.auth.getUserByName(name)
-        if user is None:
+        authcell = self.getAuthCell()
+        udef = await authcell.getUserDefByName(name)
+        if udef is None:
             return self.sendRestErr('AuthDeny', 'No such user.')
 
-        if not await user.tryPasswd(passwd):
+        if not await authcell.tryUserPasswd(name, passwd):
             return self.sendRestErr('AuthDeny', 'Incorrect password.')
 
+        iden = udef.get('iden')
         sess = await self.sess()
+        await sess.set('user', iden)
+        await sess.set('username', name)
```

Comment:
We should probably discuss that; since I am unaware of anyone hooking those events fired on the Sess handlers. 
Catching the events fired on them is non-trivial, since it requires an implementer to override potentially many different methods to grab them before they are used.